git/index/util.py

```diff
@@ -47,12 +48,21 @@ def __init__(self, file_path: PathLike) -> None:
         except OSError:
             pass
 
-    def __del__(self) -> None:
+    def __enter__(self) -> "TemporaryFileSwap":
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> bool:
         if osp.isfile(self.tmp_file_path):
             if is_win and osp.exists(self.file_path):
                 os.remove(self.file_path)
             os.rename(self.tmp_file_path, self.file_path)
-        # END temp file exists
+
+        return False
```

Comment:
Is there any reason exceptions shouldn't propagate?

Reply:
I think that `False` means it should be propageted

Reply:
Ah, true, I must have flipped bit there. Thanks for bearing with me on this one.